ext/nnstreamer/android_source/gstamcsrc.c

```diff
@@ -0,0 +1,971 @@
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("unlocking create");
+  gst_data_queue_set_flushing (priv->outbound_queue, TRUE);
+
+  return TRUE;
+}
+
+/**
+ * @brief unlock_stop function, clear the previous unlock request
+ */
+static gboolean
+gst_amc_src_unlock_stop (GstBaseSrc * src)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("stopping unlock");
+  gst_data_queue_set_flushing (priv->outbound_queue, FALSE);
+
+  return TRUE;
+}
+
+/**
+ * @brief get caps of subclass
+ */
+static GstCaps *
+gst_amc_src_get_caps (GstBaseSrc * src, GstCaps * filter)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+  GstCaps *caps = gst_caps_new_empty ();
+  GstStructure *format;
+
+  GST_OBJECT_LOCK (self);
+
+  /** width, height, and framerate were obtained from the media format */
+  format = gst_structure_new ("video/x-raw",
+      "format", G_TYPE_STRING, "NV12",  /** TODO Support other formats? */
+      "width", G_TYPE_INT, priv->width,
+      "height", G_TYPE_INT, priv->height,
+      "interlaced", G_TYPE_BOOLEAN, FALSE,
+      "pixel-aspect-ratio", GST_TYPE_FRACTION, 1, 1,
+      "framerate", GST_TYPE_FRACTION, priv->framerate, 1, NULL);
+
+  gst_caps_append_structure (caps, format);
+
+  gst_structure_free (format);
+
+  GST_OBJECT_UNLOCK (self);
+
+  return caps;
+}
+
+/**
+ * @brief set a buffer which is a head item in data queue
+ */
+static GstFlowReturn
+gst_amc_src_create (GstPushSrc * src, GstBuffer ** buffer)
+{
+  GstAMCSrc *self = GST_AMC_SRC_CAST (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+  GstDataQueueItem *item;
+
+  if (!gst_data_queue_pop (priv->outbound_queue, &item)) {
+    GST_DEBUG_OBJECT (self, "We're flushing");
+    return GST_FLOW_FLUSHING;
+  }
+
+  *buffer = GST_BUFFER (item->object);
+  g_free (item);
+
+  return GST_FLOW_OK;
+}
+
+/**
+ * @brief change state function for this element
+ * each trainsition sends the corresponding message to a looper
+ */
+static GstStateChangeReturn gst_amc_src_change_state (GstElement * element, 
+    GstStateChange transition)
+{
+  GstAMCSrc *self;
+  GstAMCSrcPrivate *priv;
+  GstStateChangeReturn ret;
+  gboolean buffer_state_change_success = TRUE;
+  gchar *dirname = NULL;
+
+  self = GST_AMC_SRC (element);
+  priv  = GST_AMC_SRC_GET_PRIVATE (self); 
+
+  switch (transition) {
+    case GST_STATE_CHANGE_PAUSED_TO_PLAYING:
+      Looper_post (priv->looper, MSG_CODEC_RESUME, self, false);
+      LOGI ("PAUSED => PLAYING");
+      break;
+    default:
+      break;
+  }
+
+  ret = GST_ELEMENT_CLASS (parent_class)->change_state (element, transition);
+
+  switch (transition) {
+    case GST_STATE_CHANGE_PLAYING_TO_PAUSED:
+      Looper_post (priv->looper, MSG_CODEC_PAUSE, self, false);
+      LOGI ("PLAYING => PAUSED");
+      break;
+    default:
+      break;
+  }
+
+  if (buffer_state_change_success == FALSE) 
```

Comment:
where this flag can be changed to false?

Reply:
Oh, I'll remove this.